### PR TITLE
[api] increase length limit of project scmsync elements

### PR DESCRIPTION
**Pull request by adrianschroeter**
We are about to run into the 255 byte limit

### CODE REVIEW

**Comment by hennevogel** [comment]
Why only for Project?

**Comment by hennevogel** [comment]
Do not merge, this will require downtime that needs to be scheduled.

**Comment by AdamMajer** [comment]
It's needed for `?onlybuild=pkgA` parameter for git based projects. For example, to implement staging in git based projects where the project cannot be altered directly since it must be in merge state.

File: src/api/db/migrate/20240705121900_change_scmsync_type.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class ChangeScmsyncType < ActiveRecord::Migration[6.1]
```
Comment (eduardoj) [inline src/api/db/migrate/20240705121900_change_scmsync_type.rb:1]:
```suggestion
class ChangeScmsyncType < ActiveRecord::Migration[7.0]
```

**Review by eduardoj — APPROVED** [review]

**Pull request merged**